src/main/kotlin/com/hibob/academy/types/types.kt

```diff
@@ -15,7 +15,9 @@ data class Owner(
     val id: UUID,
     val companyId: UUID,
     val employeeId: UUID,
-    var name: String,
+    var name: String?,
     var type: String,
-    var dateOfArrival: LocalDateTime
+    var dateOfArrival: LocalDateTime,
+    var firstName: String?,
```

Comment:
Where are you using the split of the name vs. firstName and lastName?
Could you please add some mock implementation in your functions?
Also if you split v1/v2 then there is no reason to use the same object. Since you're using a new route it doesn't break existing functionality. We usually add more params to the same object when we want to use the same route as before.

Reply:
but if i want all of my future functions, isn't its better that both will be the same?

Reply:
Depends. Sometimes we need to save both until we deprecate the old version. Sometimes we do this type of changes because we want to change some logic. So until the full new logic is in production then we need to support them both.
Depends who the client is. If it's actual customers (public api) then it's better to keep both. If it's internal then up to your team.